src/ui/Tab/ResponsiveTabs.ts

```diff
@@ -0,0 +1,178 @@
+import {$$, Dom} from '../../utils/Dom.ts';
+import {InitializationEvents} from '../../events/InitializationEvents.ts';
+import {PopupUtils, HorizontalAlignment, VerticalAlignment } from '../../utils/PopupUtils';
+
+export class ResponsiveTabs {
+  static coveoTabSection = '.coveo-tab-section';
+  
+  private static dropdownHeader: Dom;
+  private static dropdownContent: Dom;
+  private static tabSection: Dom;
+  private static searchBarElement: HTMLElement;
+  private static coveoRoot: HTMLElement;
+  
+  public static init() {
+    this.coveoRoot = <HTMLElement>document.querySelector('.CoveoSearchInterface');
+    $$(this.coveoRoot).on(InitializationEvents.afterInitialization, () => {
+      this.dropdownContent = this.buildDropdownContent();
+      this.dropdownHeader = this.buildDropdownHeader();
+      this.searchBarElement = this.getSearchBoxElement();
+      let tabSectionElement = <HTMLElement>document.querySelector(this.coveoTabSection);
+
+      if (!tabSectionElement) {
+        return;
+      }
+      this.tabSection = $$(tabSectionElement);
+
+      let manageResponsiveTabs = () => {
+        this.toggleSmallTabsIfNeeded();
+        
+        let tabSectionIsOverflowing = this.isOverflowing(this.tabSection.el);
+        if (tabSectionIsOverflowing && this.tabSection.is('.coveo-small-tab-section')) {
+          let currentTab;
+          let overflowingElements = [];
+          let tabs = this.tabSection.findAll('.CoveoTab');
+          
+          this.tabSection.el.appendChild(this.dropdownHeader.el);
+          for(let i = tabs.length - 1; i >= 0; i--) {
+            currentTab = tabs[i];
+            this.addToDropdown(currentTab);
+            if(!this.isOverflowing(this.tabSection.el)) {
+              break;
+            }
+          }
+          
+        } else if(!tabSectionIsOverflowing && this.tabSection.is('.coveo-small-tab-section') && !this.isDropdownEmpty()) {
+          let dropdownTabs = this.dropdownContent.findAll('.coveo-small-tab');
+          
+          while (!this.isOverflowing(this.tabSection.el) && !this.isDropdownEmpty()) {
+            let current = dropdownTabs.shift();
+            this.removeFromDropdown(current);
+            $$(current).insertBefore(this.dropdownHeader.el);
+          }
+          
+          if (this.isOverflowing(this.tabSection.el)) {
+            let tabs = this.tabSection.findAll('.coveo-small-tab');
+            this.addToDropdown(tabs.pop());
+          }
+          
+          if (this.isDropdownEmpty()) {
+            this.dropdownHeader.detach();
+            this.toggleSmallTabsIfNeeded();
+          }
+        }
+        
+      };
+      manageResponsiveTabs();
+      window.addEventListener('resize', <EventListener>_.debounce(manageResponsiveTabs, 200));
```

Comment:
You will also need to implement a 'nuke' handler, called when the interface is removed from the page. The nuke handler will need to tidy up all events bound to the window/document/body of the page.